docs/architecture.md

```diff
@@ -190,3 +320,13 @@ await client.query(Q1) // returns { _id: 'my-todo', _type: 'io.cozy.todos', labe
 
 Q1 retrieves the full content of 'my-todo' while Q2 only retrieves its label. When Q2 is run, 'my-todo' already exists in the store, as Q1 was previsouly run. However, the fields retrieved by Q1 for 'my-todos' but absent from Q2 are not lost. This way, when Q1 is run again, the results fetched from the store (thanks to the fetch policy) will include all the fields.
 
+The problem is when some fields are removed from a document: the store will not reflect this change and result in stale data.
+
+
+### In-memory queries evaluation
+
+Each time a change occurs in the store's `documents` collection, either through a query or an update, all the stored queries are dynamically re-evaluated.
+
+As we do not want to request the actual database each time, we do the evaluation directly from the store, in memory. To achieve this, we take advantage of the [mongo-like nature](https://docs.couchdb.org/en/stable/api/database/find.html#find-selectors) of our queries definition and use the [sift.js](https://github.com/crcn/sift.js) library to evaluate the queries against the store's `documents`.
+
+This library is able to take as input a mongo query definition and transform it into a javascript predicate. Then, all the queries are evaluated on the stores's `documents`, and efficiently updated accordingly to the results.
```

Comment:
I think that if the user is not well aware of couch and mongo, we use `mango` and `mongo` and it's certainly chaotic to understand.